geopy/geocoders/mapbox.py

```diff
@@ -0,0 +1,215 @@
+from geopy.compat import urlencode, quote, string_compare
+from geopy.exc import (
+    GeocoderQueryError,
+)
+from geopy.geocoders.base import DEFAULT_SENTINEL, Geocoder
+from geopy.location import Location
+from geopy.point import Point
+from geopy.util import logger
+
+__all__ = ("MapBox", )
+
+
+class MapBox(Geocoder):
+    """Geocoder using the Mapbox API.
+
+    Documentation at:
+        https://www.mapbox.com/api-documentation/
+
+    .. versionadded:: 1.17.0
+    """
+
+    api_path = '/geocoding/v5/mapbox.places/%(query)s.json/'
+
+    def __init__(
+            self,
+            api_key,
+            format_string=None,
+            scheme=None,
+            timeout=DEFAULT_SENTINEL,
+            proxies=DEFAULT_SENTINEL,
+            user_agent=None,
+            ssl_context=DEFAULT_SENTINEL,
+            domain='api.mapbox.com',
+    ):
+        """
+        :param str api_key: The API key required by Mapbox to perform
+            geocoding requests. API keys are managed through Mapox's account
+            page (https://www.mapbox.com/account/access-tokens).
+
+        :param str domain: base api domain for mapbox
```

Comment:
The order of params in the docstring should match the order of params of the function.